library/src/contraction/contraction_solution_impl.hpp

```diff
@@ -46,20 +46,68 @@ namespace std
 
 namespace hiptensor
 {
-    template <typename DeviceOp, typename Enabler = void>
+    template <typename DeviceOp, typename DataType, typename Enabler = void>
```

Comment:
You don't need to add DataType as a template parameter here. 
You can get the type information you need below from the typename MetaTraits<DeviceOp>::ADataT